packages/cli/src/controller/scaffoldgen-controller.ts

```diff
@@ -0,0 +1,232 @@
+    const handler: HandlerType = {
+      handler: `handle${upperFirst(event.name)}_${abiName}Log`,
+      kind: 'ethereum/LogHandler',
+      filter: {
+        topics: [event.method],
+      },
+    };
+    formattedHandlers.push(handler);
+  });
+
+  return {
+    kind: 'ethereum/Runtime',
+    startBlock: userInput.startBlock,
+    options: {
+      abi: abiName,
+      address: userInput.address,
+    },
+    assets: {
+      [abiName]: {
+        file: userInput.abiPath,
+      },
+    },
+    mapping: {
+      file: DEFAULT_HANDLER_BUILD_PATH,
+      handlers: formattedHandlers,
+    },
+  };
+}
+
+export async function generateManifest(projectPath: string, manifestPath: string, userInput: UserInput): Promise<void> {
+  try {
+    const existingManifest = (await fs.promises.readFile(path.join(projectPath, manifestPath), 'utf8')) as any;
+    const existingManifestData = parseDocument(existingManifest);
+    const clonedExistingManifestData = existingManifestData.clone();
+
+    const existingDatasource = existingManifestData.get('dataSources') as any;
+
+    // Should be for every ABI
+
+    const newDataSourcesData = existingDatasource.toJSON().concat(...[constructDatasources(userInput)]);
+    clonedExistingManifestData.set('dataSources', newDataSourcesData);
+
+    // load yaml
+    await fs.promises.writeFile(path.join(projectPath, manifestPath), clonedExistingManifestData.toString(), 'utf8');
+  } catch (e) {
+    throw new Error(e);
+  }
+}
+
+export interface handlerPropType {
+  name: string;
+  argName: string;
+  argType: string;
+}
+
+export interface abiPropType {
+  name: string;
+  handlers: handlerPropType[];
+}
+
+export function constructHandlerProps(methods: [SelectedMethod[], SelectedMethod[]], abiName: string): abiPropType {
+  const handlers: handlerPropType[] = [];
+  const [events, functions] = methods;
+
+  functions.map((fn) => {
+    const fnProp: handlerPropType = {
+      name: `handle${upperFirst(fn.name)}`,
+      argName: 'tx',
+      argType: `${upperFirst(fn.name)}Transaction`,
+    };
+    handlers.push(fnProp);
+  });
+
+  events.map((event) => {
+    const fnProp: handlerPropType = {
+      name: `handle${upperFirst(event.name)}`,
+      argName: 'log',
+      argType: `${upperFirst(event.name)}Log`,
+    };
+    handlers.push(fnProp);
+  });
+
+  return {
+    name: abiName,
+    handlers: handlers,
+  };
+}
+
+export async function generateHandlers(
+  selectedMethods: [SelectedMethod[], SelectedMethod[]],
+  projectPath: string,
+  abiPath: string
+): Promise<void> {
+  const abiProps = constructHandlerProps(selectedMethods, parseContractPath(abiPath).name);
+
+  try {
+    await renderTemplate(
+      SCAFFOLD_HANDLER_TEMPLATE_PATH,
+      path.join(projectPath, ROOT_MAPPING_DIR, 'mappingHandlers.ts'),
+      {
+        props: {
+          abis: [abiProps],
+        },
+        helper: {upperFirst},
+      }
+    );
+  } catch (e) {
+    console.error(`unable to generate scaffold. ${e.message}`);
```

Comment:
When writing error messages that are presented to users try and make them useful:
- They shouldn't need to understand internal workings or terminology
- Try and offer a solution or some way to problem solve